File: 2745-construct-the-longest-new-string/2745-construct-the-longest-new-string.cpp
class Solution {
public:
    int longestString(int x, int y, int z) {
        int ans = 0;
        if (x == y)
            ans = 2 * min(x,y);
        else
            ans = 2 * min(x,y) + 1;
            
        ans += z;

        return ans * 2;
    }
};

File: 2745-construct-the-longest-new-string/2745-construct-the-longest-new-string.rs
impl Solution {
    pub fn longest_string(x: i32, y: i32, z: i32) -> i32 {
        let mut ans = 0;
        if x == y {
            ans = 2 * x.min(y);
        } else {
            ans = 2 * x.min(y) + 1;
        }

        ans += z;

        ans * 2
    }
}
